mlx/directives/item_pie_chart_directive.py

```diff
@@ -72,7 +72,11 @@ def perform_replacement(self, app, collection):
                            self['document'], self['line'])
         p_node = nodes.paragraph()
         p_node += nodes.Text(statistics)
-        top_node += self.build_pie_chart(chart_labels, env)
+        if chart_labels:
+            top_node += self.build_pie_chart(chart_labels, env)
+        else:
+            report_warning("item-piechart does not contain any items: no image will be generated",
```

Comment:
I looked at our internal example and warning in our case would mean: "Great job, there is no failures left for you to fix." So, can we make this somehow optional, with some flag to the directive?

Reply:
A new flag like `:allowempty:`? Can do that.

Reply:
yeah sounds reasonable. And also put the hint in the warning. Like, if this is intended use :allowempty: attribute to your item-piechart directive.

Reply:
We decided to remove the new warning because the plugin doesn't report a warning when an item-matrix is empty, for example.